build: plumb COCKROACH_PROPOSER_EVALUATED_KV env var

tbg opened the pull request:
This should make it easy to run a nightly build for which tests are run using
proposer-evaluated KV.

Updates #10454.

<!-- Reviewable:start -->

---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/10455)
<!-- Reviewable:end -->

tschottdorf commented:
My current impression is that the best way to set up the nightly is to trigger the `GitHub CI` job with `env.COCKROACH_PROPOSER_EVALUATED_KV=true`. Please let me know if that's a bad idea.

tamird commented:
What are you trying to achieve? Do you want a single nightly run of the tests with prop-evaluated kv, or do you want stress with that configuration?

---

Review status: 0 of 5 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10455#-:-KVjdJN0bxA7_iWoh_2J:b6lijd7)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
The former, with an option of the latter (but that would presumably use your stress-meta).

jordanlewis commented:
I think this strategy makes sense. The main pitfall you'll run into is that there's no UI option to trigger with a set of environment variables - you need to use an API call. It shouldn't be too hard, though - make a nightly job that hits that API endpoint. http://stackoverflow.com/questions/10007874/trigger-option-to-set-specific-build-parameters

---

Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[build/teamcity-acceptance.sh, line 7 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10455#-KVjmjP0LF1mUXyhK3bK:-KVjmjP0LF1mUXyhK3bL:bvfkl5i) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b407ae551bf489d3e101efaa14af27bb6dbe7c2/build/teamcity-acceptance.sh#L7)):*
> ```Shell
> build/builder.sh go test -v -c -tags acceptance ./pkg/acceptance
> cd pkg/acceptance
> COCKROACH_PROPOSER_EVALUATED_KV="${COCKROACH_PROPOSER_EVALUATED_KV-}" ../../acceptance.test -nodes 3 -l ../../artifacts/acceptance -test.v -test.timeout 10m 2>&1 | go-test-teamcity
> ```

I don't think you actually need to do this, since we're not running this command from within the builder. I would suggest `echo`ing the value of this variable on the line above or something to make it clear in the build log what's going on.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10455#-:-KVjnZ9udQXbaiX64FQB:b8g2w3n)*
<!-- Sent from Reviewable.io -->

tamird commented:
There is a UI option, it's just not very ergonomic. <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[build/teamcity-acceptance.sh, line 7 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10455#-KVjmjP0LF1mUXyhK3bK:-KVjowQjkT3GIIuznxlC:b6z3pfu) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b407ae551bf489d3e101efaa14af27bb6dbe7c2/build/teamcity-acceptance.sh#L7)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary>
> I don't think you actually need to do this, since we're not running this command from within the builder. I would suggest `echo`ing the value of this variable on the line above or something to make it clear in the build log what's going on.
</details>
+1

don't echo this variable unless you plan to also echo the entire env

---

*[pkg/cmd/github-pull-request-make/main.go, line 169 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10455#-KVjonJgTkRAZMf5SnAo:-KVjdBrbRkwIHqYlqwoP:b-e2rosd) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b407ae551bf489d3e101efaa14af27bb6dbe7c2/pkg/cmd/github-pull-request-make/main.go#L169)):*
> ```Go
> 				fmt.Sprintf("PKG=./%s", name),
> 				fmt.Sprintf("TESTS=%s", tests),
> 				fmt.Sprintf("COCKROACH_PROPOSER_EVALUATED_KV=%t", envutil.EnvOrDefaultBool("COCKROACH_PROPOSER_EVALUATED_KV", false)),
> ```

remove this, since it will literally not do anything (PR builds don't have this env var set, unless you plan to set it?)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10455#-:-KVjp1rw6u-b2HvYyilf:b-r1giwy)*
<!-- Sent from Reviewable.io -->

tamird commented:
Oh, there might not be a UI option for a scheduled trigger with specific parameters?

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10455#-:-KVjp7Ca16EmKTVjky7T:bebpqc9)*
<!-- Sent from Reviewable.io -->

jordanlewis commented:
Yes, that's what I'm saying.

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10455#-:-KVjrRBwnjTq65s7A3Ms:b-nlmq0v)*
<!-- Sent from Reviewable.io -->

tbg committed:
build: plumb COCKROACH_PROPOSER_EVALUATED_KV env var

This should make it easy to run a nightly build for which tests are run using
proposer-evaluated KV.

Updates #10454.

tschottdorf commented:
OK, plumbing this only for the main test and testrace invocation (and implicitly for `acceptance`). I also let `cockroach` print a message when it's running with proposer-evaluated KV; that way test failures should be easy to recognize. PTAL and then I'll try to set up the CI.

---

Review status: 2 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[build/teamcity-acceptance.sh, line 7 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10455#-KVjmjP0LF1mUXyhK3bK:-KVkiFkTAXpI2C6VtHZe:bss4a2b) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b407ae551bf489d3e101efaa14af27bb6dbe7c2/build/teamcity-acceptance.sh#L7)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> +1
> 
> don't echo this variable unless you plan to also echo the entire env
</details>
Removed.

---

*[pkg/cmd/github-pull-request-make/main.go, line 169 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10455#-KVjonJgTkRAZMf5SnAo:-KVkiHYCd4dTYEW6_53B:b-4xolqy) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b407ae551bf489d3e101efaa14af27bb6dbe7c2/pkg/cmd/github-pull-request-make/main.go#L169)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> remove this, since it will literally not do anything (PR builds don't have this env var set, unless you plan to set it?)
</details>
Done. Not planning to set this, just plumbed a little bit too much.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10455#-:-KVkiKam_QmkFdczqCiB:bmswl76)*
<!-- Sent from Reviewable.io -->
